QuestionBot/QuestionBot.UnitTests/Model/MessageEmitterTests.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Collections.Generic;
+using NUnit.Framework;
+using QuestionBot.Model;
+using Moq;
+
+namespace QuestionBot.UnitTests.Model {
+    [TestFixture]
+    internal class MessageEmitterTests {
+        private MessageEmitter _testEmitter;
+        private Mock<IMessageListener> _testListener;
+        private Mock<IConsole> _testConsole;
+        private string _exitCommand;
+
+
+        [SetUp]
+        public void Setup() {
+            _testListener = new Mock<IMessageListener>( MockBehavior.Strict );
+            _testConsole = new Mock<IConsole>();
+            _exitCommand = "/exitQuestionBot";
+        }
+
+        [Test]
+        public void Receiving_Message_Will_Notify_Listener() {
+            const string message = "/question What is 1+1?";
+            const string output = "Thank you";
+            const string nullOutput = null;
+
+
+            _testConsole.SetupSequence( x => x.ReadLine() )
+                .Returns( message )
+                .Returns( _exitCommand );
+
+            _testListener.Setup( x => x.ReceiveMessage( message ) ).Returns( output );
+            _testListener.Setup( x => x.ReceiveMessage( _exitCommand ) ).Returns( nullOutput );
+
+            _testEmitter = new MessageEmitter( _testConsole.Object );
+            _testEmitter.Add( _testListener.Object );
+            _testEmitter.Start();
+
+            _testListener.Verify( x => x.ReceiveMessage( message ), Times.Exactly( 1 ) );
+            _testListener.Verify( x => x.ReceiveMessage( _exitCommand ), Times.Exactly( 1 ) );
+            _testConsole.Verify( x => x.WriteLine( output ), Times.Exactly( 1 ) );
+        }
+
+        [Test]
+        public void Receiving_Message_With_No_Listeners_Will_Not_Call_ReceiveMessage() {
+            const string message = "/question What is 1+1?";
+            const string output = "Thank you";
+
+            _testConsole.SetupSequence( x => x.ReadLine() )
+                .Returns( message )
+                .Returns( _exitCommand );
+
+            _testListener.Setup( x => x.ReceiveMessage( message ) ).Returns( output );
```

Comment:
If you don't call `Setup`, then we also don't have to `Verify` because `MockBehaviour.Strict` means that the Mock will throw if you call a method that wasn't set up. This makes the test much cleaner.